interpreter/terminal_interface/magic_commands.py

```diff
@@ -23,38 +32,45 @@ def handle_undo(self, arguments):
         removed_messages = self.messages[last_user_index:]
         self.messages = self.messages[:last_user_index]
 
-    print("") # Aesthetics.
+    print("")  # Aesthetics.
 
     # Print out a preview of what messages were removed.
     for message in removed_messages:
-      if 'content' in message and message['content'] != None:
-        display_markdown_message(f"**Removed message:** `\"{message['content'][:30]}...\"`")
-      elif 'function_call' in message:
-        display_markdown_message(f"**Removed codeblock**") # TODO: Could add preview of code removed here.
-    
-    print("") # Aesthetics.
+        if 'content' in message and message['content'] != None:
+            display_markdown_message(
+                f"**Removed message:** `\"{message['content'][:30]}...\"`")
+        elif 'function_call' in message:
+            # TODO: Could add preview of code removed here.
+            display_markdown_message(f"**Removed codeblock**")
+
+    print("")  # Aesthetics.
+
 
 def handle_help(self, arguments):
     commands_description = {
-      "%debug [true/false]": "Toggle debug mode. Without arguments or with 'true', it enters debug mode. With 'false', it exits debug mode.",
-      "%reset": "Resets the current session.",
-      "%undo": "Remove previous messages and its response from the message history.",
-      "%save_message [path]": "Saves messages to a specified JSON path. If no path is provided, it defaults to 'messages.json'.",
-      "%load_message [path]": "Loads messages from a specified JSON path. If no path is provided, it defaults to 'messages.json'.",
-      "%tokens [prompt]": "Calculate the tokens used by the current conversation's messages and estimate their cost and optionally calculate the tokens and estimated cost of a `prompt` if one is provided.",
-      "%help": "Show this help message.",
+        "%debug [true/false]": "Toggle debug mode. Without arguments or with 'true', it enters debug mode. With 'false', it exits debug mode.",
+        "%reset": "Resets the current session.",
+        "%undo": "Remove previous messages and its response from the message history.",
+        "%save_message [path]": "Saves messages to a specified JSON path. If no path is provided, it defaults to 'messages.json'.",
+        "%load_message [path]": "Loads messages from a specified JSON path. If no path is provided, it defaults to 'messages.json'.",
+        "%tokens [prompt]": "Calculate the tokens used by the current conversation's messages and estimate their cost and optionally calculate the tokens and estimated cost of a `prompt` if one is provided.",
+        "%help": "Show this help message.",
+        "%upload": "open a File Dialog, and select a file to upload to the container. only used when using containerized code execution",
+        "%upload folder": "same as upload command, except you can upload a folder instead of just a file.",
+        "%upload file": "same as upload command, except you can upload a file.",
```

Comment:
When using `%upload file` it still seems to open the `Do you want to upload a folder or a file?" window.

Maybe we should just describe the `%upload` magic command for now and revisit the `file` and `folder` functionality in a separate PR after we are able to get this one ready and merged in?

Reply:
I think i will just remove file and folder for the command. it will still work, it is only since you cannot open a file dialog which can select files and folders, so I need to know which to open. but when using it via cli its not needed.

Reply:
Oh, I think I misunderstood the purpose of them.

I’ll retest them in a bit and see if they open the correct file/folder selector.

We might just need to add some docs about usage to avoid confusion.